test/e2e/flowaggregator_test.go

```diff
@@ -1753,14 +1765,24 @@ func deletePerftestServices(t *testing.T, data *TestData) {
 	}
 }
 
-func addLabelToPerftestPods(t *testing.T, data *TestData, label string) {
-	perftestPods, err := data.clientset.CoreV1().Pods(data.testNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "app=iperf"})
-	require.NoError(t, err, "Error when getting perftest Pods")
-	for i := range perftestPods.Items {
-		pod := &perftestPods.Items[i]
-		pod.Labels["targetLabel"] = label
-		_, err = data.clientset.CoreV1().Pods(data.testNamespace).Update(context.TODO(), pod, metav1.UpdateOptions{})
-		require.NoErrorf(t, err, "Error when adding label to %s", pod.Name)
+func addLabelToTestPods(t *testing.T, data *TestData, label string, podNames []string) {
+	for _, podName := range podNames {
+		testPod, err := data.clientset.CoreV1().Pods(data.testNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		require.NoErrorf(t, err, "Error when getting Pod %s in %s", testPod, data.testNamespace)
+		testPod.Labels["targetLabel"] = label
+		_, err = data.clientset.CoreV1().Pods(data.testNamespace).Update(context.TODO(), testPod, metav1.UpdateOptions{})
+		require.NoErrorf(t, err, "Error when adding label to %s", testPod.Name)
+		err = wait.Poll(defaultInterval, timeout, func() (bool, error) {
+			pod, err := data.clientset.CoreV1().Pods(data.testNamespace).Get(context.TODO(), testPod.Name, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, fmt.Errorf("error when getting Pod '%s': %w", pod.Name, err)
+			}
+			return pod.Labels["targetLabel"] == label, nil
+		})
+		require.NoErrorf(t, err, "Error when verifying the label on %s", testPod.Name)
```

Comment:
I'm going to merge the PR as is, but I don't think this is useful. It should not be possible to have a successful `Update` call, yet see the old object when calling `Get` afterwards.